src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/hot_node/ThreadMetricsRca.java

```diff
@@ -0,0 +1,269 @@
+
+/**
+ * Analyzes thread_blocked_time and thread_waited_time metrics for transport threads and publishes
+ * the following data: 1. Number of threads which are blocked for more than 10 seconds in past
+ * minute. 2. Number of threads which are in waiting/timed-waiting state for more than 10 seconds in
+ * past minute. 3. Maximum value of thread_blocked_time for a single thread in the past minute. 4.
+ * Maximum value of thread_waited_time for a single thread in the past minute.
+ */
+public class ThreadMetricsRca extends Rca<ResourceFlowUnit<HotNodeSummary>> {
+
+    private static final Logger LOG = LogManager.getLogger(ThreadMetricsRca.class);
+    private final int rcaPeriod;
+    private final Thread_Blocked_Time threadBlockedTime;
+    private final Thread_Waited_Time threadWaitedTime;
+    private final ThreadMetricsSlidingWindow blockedTransportThreadsSlidingWindow;
+    private final ThreadMetricsSlidingWindow waitedTransportThreadsSlidingWindow;
+    private final Clock clock;
+
+    public ThreadMetricsRca(
+            final Thread_Blocked_Time threadBlockedTime,
+            Thread_Waited_Time threadWaitedTime,
+            final int rcaPeriodInSeconds) {
+        super(rcaPeriodInSeconds);
+        this.rcaPeriod = rcaPeriodInSeconds;
+        this.threadBlockedTime = threadBlockedTime;
+        this.threadWaitedTime = threadWaitedTime;
+        blockedTransportThreadsSlidingWindow = new ThreadMetricsSlidingWindow();
+        waitedTransportThreadsSlidingWindow = new ThreadMetricsSlidingWindow();
+        this.clock = Clock.systemUTC();
+    }
+
+    @Override
+    public void generateFlowUnitListFromWire(FlowUnitOperationArgWrapper args) {
+        final List<FlowUnitMessage> flowUnitMessages =
+                args.getWireHopper().readFromWire(args.getNode());
+        final List<ResourceFlowUnit<HotNodeSummary>> flowUnitList = new ArrayList<>();
+        LOG.debug("rca: Executing fromWire: {}", this.getClass().getSimpleName());
+        for (FlowUnitMessage flowUnitMessage : flowUnitMessages) {
+            flowUnitList.add(ResourceFlowUnit.buildFlowUnitFromWrapper(flowUnitMessage));
+        }
+        setFlowUnits(flowUnitList);
+    }
+
+    @Override
+    public void readRcaConf(RcaConf conf) {}
+
+    @Override
+    public ResourceFlowUnit<HotNodeSummary> operate() {
+        try {
+            long currentTimeMillis = this.clock.millis();
+            LOG.debug("ThreadMetricsRca run at {}", currentTimeMillis);
+            collateThreadMetricData(
+                    currentTimeMillis, threadBlockedTime, blockedTransportThreadsSlidingWindow);
+            collateThreadMetricData(
+                    currentTimeMillis, threadWaitedTime, waitedTransportThreadsSlidingWindow);
+
+            int blockedTransportThreadCount =
+                    blockedTransportThreadsSlidingWindow.getCountExceedingThreshold(5d);
+            PerformanceAnalyzerApp.READER_METRICS_AGGREGATOR.updateStat(
```

Comment:
[NIT] Define 5d as constant/setting at class level. Plus shouldn't this have been 10 seconds as mentioned in javadoc above?